compiler/modules/front.syntax/token.bal

```diff
@@ -487,6 +487,30 @@ public readonly class SourceFile {
         return unpackPosition(pos);
     }
 
+    public function lineContent(Position pos) returns string {
+        int lineNum = self.lineColumn(pos)[0];
```

Comment:
This should be factored out into a scanLineToString function in scan.bal.

Create an array of strings and then join them at the end, rather than using `+=`.

This needs some tests (maybe include `in testParserOnTestSuite`): scanLineToString should always be exactly equal to the original string.

Reply:
fixed